department-of-veterans-affairs/caseflow
# Will/appeals 34277 Pagination - with fixed merge conflicts

Resolves [APPEALS-34277 Intake Form Step 2: Review Tasks & Appeals - Pagination](https://jira.devops.va.gov/browse/APPEALS-34277)
Resolves [Test Coverage: Review Tasks & Appeals - Tasks related to an existing Appeal](https://jira.devops.va.gov/browse/APPEALS-32888)
# Description

- Added pagination to Existing appeal table
- Added appeals to `multi_correspondences.rb` seed file
- Added redux state management to handle checkbox selections
- Refactored the case details page links to open in new tab
- Added RSpec feature test

## Acceptance Criteria

## Testing Plan

1. Go to [Jira Issue/Test Plan Link](https://jira.devops.va.gov/browse/JIRA-12345) or list them below

- Navigate to voyager.caseflowdemo.com/queue/correspondence
- Click on one of the veterans
- Click on "Create a record"
- Click on "Continue"
  - Expected result: There is a section entitled: "Tasks related to an existing Appeal"
- Click on the radio button "Yes"
  - Expected result: A table appears with showing 5 appeals
  - The table has a pagination summary (Viewing 1-5 of XXX total)
  - The table has the following headers:
    - Docket
    - Appellant Name
    - Status
    - Types
    - Number of Issues
    - Decision Date
    - Assigned To
  - Each row has a checkbox
  - There are pagination buttons below the table
    - The "Previous" button is grey and disabled on page 1
    - The "Next" button is grey and disabled on the last page
  - The values under the Docket header are clickable links
- Click on one of the links under Docket
  - Expected result: A new tab opens up to that appeal's case details page
- Go back to the intake form tab
- Check some checkboxes
- Click on the pagination buttons
  - Expected result: The checkbox values persist through page changes
  - The pagination summary updates correctly
- Click on the "No" radio button above the table
- Click on the "Yes" radio button above the table
  -  Any previously checked checkboxes are now cleared


# Frontend
![Screenshot 2023-11-20 at 3 03 19 PM](https://github.com/department-of-veterans-affairs/caseflow/assets/93014155/22ea1731-0eb5-4e79-8c2d-ffb48c72e6f6)



## Storybook Story
*For Frontend (Presentation) Components*

# Backend
## Database Changes
*Only for Schema Changes*


## Integrations: Adding endpoints for external APIs

# Best practices
## Code Documentation Updates

## Tests
### Test Coverage
Did you include any test coverage for your code? Check below:

### Code Climate
Your code does not add any new code climate offenses? If so why?

## Monitoring, Logging, Auditing, Error, and Exception Handling Checklist

### Monitoring

### Logging

### Auditing

### Error Handling

### Exception Handling
